website/addons/onedrive/model.py

```diff
@@ -0,0 +1,258 @@
+            token = self._refresh_token(self.account.oauth_key, self.account.refresh_token)
+            self.account.oauth_key = token['access_token']
+            self.account.refresh_token = token['refresh_token']
+            self.account.expires_at = datetime.utcfromtimestamp(token['expires_at'])
+            self.account.save()
+
+    def _needs_refresh(self):
+        if self.account.expires_at is None:
+            return False
+        return (self.account.expires_at - datetime.utcnow()).total_seconds() < settings.REFRESH_TIME
+
+class OneDriveUserSettings(AddonOAuthUserSettingsBase):
+    """Stores user-specific onedrive information
+    """
+    oauth_provider = OneDrive
+    serializer = OneDriveSerializer
+#     myBase = AddonOAuthUserSettingsBase
+
+
+class OneDriveNodeSettings(StorageAddonBase, AddonOAuthNodeSettingsBase):
+
+    oauth_provider = OneDrive
+    serializer = OneDriveSerializer
+
+    foreign_user_settings = fields.ForeignField(
+        'onedriveusersettings', backref='authorized'
+    )
+    folder_id = fields.StringField(default=None)
+    onedrive_id = fields.StringField(default=None)
+    folder_name = fields.StringField()
+    folder_path = fields.StringField()
+
+    _folder_data = None
+
+    _api = None
+
+    @property
+    def api(self):
+        """authenticated ExternalProvider instance"""
+        if self._api is None:
+            self._api = OneDrive(self.external_account)
+        return self._api
+
+    @property
+    def display_name(self):
+        return '{0}: {1}'.format(self.config.full_name, self.folder_id)
+
+    @property
+    def has_auth(self):
+        """Whether an access token is associated with this node."""
+        return bool(self.user_settings and self.user_settings.has_auth)
+
+    @property
+    def complete(self):
+        return bool(self.has_auth and self.user_settings.verify_oauth_access(
+            node=self.owner,
+            external_account=self.external_account,
+        ))
+
+    def fetch_folder_name(self):
+        self._update_folder_data()
+        return self.folder_name.replace('All Files', '/ (Full OneDrive)')
+
+    def fetch_full_folder_path(self):
+        self._update_folder_data()
+        return self.folder_path
+
+    def _update_folder_data(self):
+        if self.folder_id is None:
+            return None
+
+        logger.debug('self::' + repr(self))
+        #request.json.get('selected')
+
+        if not self._folder_data:
+
+            #self.folder_name = self.folder_id #request.json.get('selected')['name']  # 'Test TBD' #self._folder_data['name']
+            self.path = self.folder_name
+#             self.folder_path = '/'.join(
+#                 [x['name'] for x in self._folder_data['path_collection']['entries']]
+#                 + [self._folder_data['name']]
+#             )
+            self.save()
+
+    def set_folder(self, folder, auth):
+        self.onedrive_id = folder['id']
+        self.folder_id = folder['name'] #folder['id']
```

Comment:
Can you please explain the meanings behind 'id' and 'name' in this context? It seems to me it might make sense to either:
 
- redefine `folder_id` as a property method proxying `onedrive_id`
- get rid of `onedrive_id` in favor of `folder_id`